apps/admin_app/assets/js/components/CreatePromotions.tsx

```diff
@@ -0,0 +1,369 @@
+        addedActions.splice(index,1)
+        this.setState({addedActions:addedActions})
+    }
+
+    renderRuleOptions=()=>{
+        switch(this.state.selectedRule){
+            case "Elixir.Snitch.Data.Schema.PromotionRule.OrderTotal":
+                return(
+                <div>
+                    Lower Range:
+                    <InputNumber placeholder="Lower Range" onChange={(value)=>{this.setState({lowRange:value})}} />
+                    <br/>
+                    Upper Range:
+                    <InputNumber placeholder="Upper Range" onChange={(value)=>{this.setState({upRange:value})}}/>
+                    <Button icon="save" onClick={()=>{this.setState({addRule:false,selectedRule:""});this.handleSaveRule({"name":"Order Item Total","value":{",LowerRange":this.state.lowRange,",UpperRange":this.state.upRange}});}}>Save</Button>
+                </div>
+                )
+            case "Elixir.Snitch.Data.Schema.PromotionRule.Product":
+            const Option = Select.Option;
+                return(
+                    <div>
+                        Products:
+                        <Select mode="multiple" onChange={(selectedProducts)=>{this.setState({selectedProducts:selectedProducts})}}>
+                        <Option value="shoes">
+                         shoes
+                        </Option>
+                        <Option value="tshirts">
+                            tshirts
+                        </Option>
+                        <Option value="watches">
+                            watches
+                        </Option>
+
+                        </Select>
+
+                        Match Policy:
+                        <Select onChange={(productMatchPolicy)=>{this.setState({productMatchPolicy:productMatchPolicy})}}>
+                            <Option value = "all">All</Option>
+                            <Option value="any">Any</Option>
+                            <Option value="none">None</Option>
+                        </Select> 
+                        <Button icon="save" onClick={()=>{this.setState({addRule:false,selectedRule:""});this.handleSaveRule({"name":"Product Rule","value":{",Categories":this.state.selectedProducts,",Match Policy":this.state.productMatchPolicy}});}}>Save</Button>
+                        
+                    </div>
+                )
+            default:
+                return null
+        }
+        
+    }
+
+    handleCalculator=(calc)=>{
+        console.log("calcc",calc)
+        this.setState({selectedCalc:calc})
+        var url="http://localhost:4000/api/promo-calc-prefs"
+        var data={"calculator":calc}
+        this.fetchPOST(url,data).then(response=>{}).catch(error=>console.error('Error',error));
+    }
+
+    renderCalculatorOptions=()=>{
+        console.log("calc",this.state.selectedCalc)
+        switch(this.state.selectedCalc){
+            case "Elixir.Snitch.Domain.Calculator.FlatRate":
```

Comment:
@arjun289 this is not the best approach, can we please think of a better approach?

Reply:
@ashish173 I totally agree. This has to be dynamic, the API for getting preferences for a calculator is such that it can be rendered without hardcoding it like this.